### PR TITLE
chore: refactor promiseRecord

### PR DESCRIPTION
**Pull request by runspired**
tiny bit of cleanup to the `store.js` file as we prepare to refactor it more

### CODE REVIEW

File: packages/store/addon/-private/utils/promise-record.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Get the materialized model from the internalModel/promise
+ * that returns an internal model and return it in a promiseObject.
+ *
+ * Useful for returning from find methods
+ *
+ * @private
+ * @param internalModelPromise
+ * @param label
+ */
+export default function promiseRecord(internalModelPromise: Promise<InternalModel>, label: string) {
```
Comment (runspired) [inline packages/store/addon/-private/utils/promise-record.ts]: `InternalModel` is the wrong type here, we need the `Record` type (which I've created elsewhere, will find)

Comment (igorT) [reply packages/store/addon/-private/utils/promise-record.ts]: Can you consult @mike-north on this

Comment (runspired) [reply packages/store/addon/-private/utils/promise-record.ts]: I can//will but it shouldn't block this as I don't add a return type in this PR

File: packages/store/addon/-private/utils/promise-record.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// these proxies
+export type PromiseProxy<T> = Promise<T>;
+
+/**
+ * Get the materialized model from the internalModel/promise
+ * that returns an internal model and return it in a promiseObject.
+ *
+ * Useful for returning from find methods
+ *
+ * @private
```
Comment (runspired) [inline packages/store/addon/-private/utils/promise-record.ts]: This should be @internal

File: packages/store/addon/-private/utils/promise-record.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * @param internalModelPromise
+ * @param label
```
Comment (runspired) [inline packages/store/addon/-private/utils/promise-record.ts]: don't need these params unless they have more docs

**Pull request merged**